programs/marginfi/src/state/marginfi_account.rs

```diff
@@ -140,10 +140,35 @@ impl<'a> BankAccountWithPriceFeed<'a> {
             .collect::<Result<Vec<_>>>()
     }
 
+    #[cfg(feature = "client")]
+    pub fn load_from_map(
+        marginfi_account: &'a MarginfiAccount,
+        banks: &std::collections::HashMap<Pubkey, Bank>,
+        price_feeds: &std::collections::HashMap<Pubkey, OraclePriceFeedAdapter>,
+    ) -> MarginfiResult<Vec<BankAccountWithPriceFeed<'a>>> {
+        marginfi_account
+            .lending_account
+            .balances
+            .iter()
+            .filter(|balance| balance.active)
+            .enumerate()
+            .map(|(_, balance)| {
+                let bank = banks.get(&balance.bank_pk).unwrap();
+                let price_feed = price_feeds.get(&bank.config.oracle_keys[0]).unwrap();
+                Ok(BankAccountWithPriceFeed {
+                    bank: Box::new(*bank),
+                    price_feed: Box::new(price_feed.clone()),
+                    balance,
+                })
+            })
+            .collect::<Result<Vec<_>>>()
+    }
+
     #[inline(always)]
-    pub fn calc_weighted_assets_and_liabilities_values(
+    pub fn calc_assets_and_liabilities_values(
         &self,
-        weight_type: WeightType,
+        weight_type: Option<WeightType>,
+        current_timestamp: i64,
```

Comment:
Why did we add the `current_timestamp` to the fn signature?

its not being used anywhere as far as I can see?

Reply:
2 reasons:

- **reason it's there**: passing the timestamp from the top rather than making the clock sysvar internal call was the fastest way to share accounting functionality with a client, as it was previously a runtime-only dependency.
- **reason it looks unused**: you wiped part of those changes during the merge you made for switchboard support + refactor, but did not remove those lingering bits. A side effect of those incoming commits was to enable client sharing as well (because we moved away from requiring the timestamp in `get_price()` altogether), so it looks like this timestamp was added for no reason if you just look at the overall diff.

I'll reiterate that it is not advised looking at this branch in its current state 😁